Foundation/NSObjCRuntime.swift

```diff
@@ -266,22 +266,143 @@ public func NSStringFromClass(_ aClass: AnyClass) -> String {
     neither stable nor human-readable.
  */
 public func NSClassFromString(_ aClassName: String) -> AnyClass? {
-    let aClassNameWithPrefix : String
-    let components = aClassName._bridgeToObjectiveC().components(separatedBy: ".")
-    
-    switch components.count {
-    case 1:
-        guard !aClassName.hasPrefix("_Tt") else {
-            NSLog("*** NSClassFromString(\(aClassName)): cannot yet decode mangled class names")
+    switch aClassName {
```

Comment:
the only other option here is to use a dictionary but this is a work-around for a bug that seems to be in `swift_getTypeByName` in the runtime library for swift or somehow in the metadata records that gets stripped out in release builds.

It seems that the failure occurs only with a two level deep inheritance:

```
// assuming module named test
class A { }
class B: A { }
class C: B { }

assert(_typeByName("test.A") != nil)
assert(_typeByName("test.B") != nil)
assert(_typeByName("test.C") != nil) // fails here
```

I need to understand a bit more to file the appropriate bugs for this.